src/popup/pages/SignMessage.vue

```diff
@@ -39,7 +39,36 @@ export default defineComponent({
         const message = isHexEncodedMessage ? Buffer.from(rawMessage, 'hex') : rawMessage;
         const displayMessage = message?.toString();
         const { host, href } = callbackOrigin.value || {} as any;
+        const jwt = route.query.jwt?.toString() === 'true';
 
+        if (jwt) {
+          const messageObject = JSON.parse(message as string);
+
+          await openModal(MODAL_MESSAGE_SIGN, {
```

Comment:
We would need either more info on existing modal or even new modal to be more user friendly for this action.

Reply:
agree

Reply:
adjusted